src/hooks/useCommands.js

```diff
@@ -0,0 +1,21 @@
+import Fuse from 'fuse.js'
+
+const useCommands = (commands, precision = 0.4) => {
+	commands = !!commands
+		? Object.entries(commands)?.reduce((acc, [key, value]) => ({ [key.toLocaleLowerCase()]: value }), {})
+		: {}
+
+	const triggerCommand = (input) => {
+		const fuse = new Fuse(Object.keys(commands), { includeScore: true, ignoreLocation: true })
```

Comment:
Shouldn't this be memoized? What would be the performance impact if I ran, say, 5 commands in a row?

Reply:
You'll be right as soon as the component handles continuous recognition. For now, it responds to one command per session only (I admit it is a great limit to the use of commands though)